src/lint/approaches_and_articles.nim

```diff
@@ -0,0 +1,88 @@
+import std/[json, strformat, strutils]
+import ".."/helpers
+import "."/validators
+
+type
+  DirKind = enum
+    dkApproaches = ".approaches"
+    dkArticles = ".articles"
+
+proc hasValidIntroduction(data: JsonNode, path: Path): bool =
+  const k = "introduction"
+  if hasObject(data, k, path):
+    let d = data[k]
+    let checks = [
+      hasArrayOfStrings(d, "authors", path, k, uniqueValues = true),
+      hasArrayOfStrings(d, "contributors", path, k, isRequired = false),
+    ]
+    result = allTrue(checks)
+
+proc isValidApproachOrArticle(data: JsonNode, context: string,
+                              path: Path): bool =
+  if isObject(data, context, path):
+    let checks = [
+      hasString(data, "uuid", path, context, checkIsUuid = true),
+      hasString(data, "slug", path, context, checkIsKebab = true),
+      hasString(data, "title", path, context, maxLen = 255),
+      hasString(data, "blurb", path, context, maxLen = 280),
+      hasArrayOfStrings(data, "authors", path, context, uniqueValues = true),
+      hasArrayOfStrings(data, "contributors", path, context,
+                        isRequired = false),
+    ]
+    result = allTrue(checks)
+
+proc isValidConfig(data: JsonNode, path: Path, dk: DirKind): bool =
+  if isObject(data, jsonRoot, path):
+    let k = dk.`$`[1..^1] # Remove dot.
+    let checks = [
+      if dk == dkApproaches: hasValidIntroduction(data, path) else: true,
+      hasArrayOf(data, k, path, isValidApproachOrArticle, isRequired = false),
+    ]
+    result = allTrue(checks)
+
+proc isConfigMissingOrValid(dir: Path, dk: DirKind): bool =
+  result = true
+  let configPath = dir / $dk / "config.json"
```

Comment:
If there is no `config.json` but there is an `.approaches/introduction.md` or there are sub-directories in the `.approaches`/`.articles` directory, it would be nice to also have an error message.